modules/apps/transfer/keeper/migrations.go

```diff
@@ -107,8 +71,41 @@ func (m Migrator) MigrateTotalEscrowForDenom(ctx sdk.Context) error {
 	return nil
 }
 
-func equalTraces(dtA, dtB types.DenomTrace) bool {
-	return dtA.BaseDenom == dtB.BaseDenom && dtA.Path == dtB.Path
+// MigrateDenomTraceToDenom migrates storage from using DenomTrace to Denom.
+func (m Migrator) MigrateDenomTraceToDenom(ctx sdk.Context) error {
+	var (
+		denoms      []types.Denom
+		denomTraces []types.DenomTrace
+	)
+	m.keeper.iterateDenomTraces(ctx,
+		func(dt types.DenomTrace) (stop bool) {
+			// convert denomTrace to denom
+			denom := types.ExtractDenomFromFullPath(dt.GetFullDenomPath())
+			err := denom.Validate()
+			if err != nil {
+				panic(err)
+			}
+
+			if dt.IBCDenom() != denom.IBCDenom() {
```

Comment:
in theory unreachable, but provides a nice defense mechanism in case we miss a step. We absolutely do not watch to be generating different sdk coin denoms after this migration, so this is just for peace of mind

Reply:
https://github.com/cosmos/ibc-go/pull/6453/commits/2f8cb7b65b7b706d1e56cef5cd21e9489a9324fb